src/jukebox/components/playermpd/__init__.py

```diff
@@ -0,0 +1,305 @@
+        elif mode == 'single':
+            repeat = 1
+            single = 1
+        else:
+            repeat = 0
+            single = 0
+
+        self.mpd_retry_with_mutex(self.mpd_client.repeat, repeat)
+        self.mpd_retry_with_mutex(self.mpd_client.single, single)
+        return None
+
+    @plugs.tag
+    def get_current_song(self, param):
+        return self.mpd_status
+
+    @plugs.tag
+    def map_filename_to_playlist_pos(self, filename):
+        # self.mpd_client.playlistfind()
+        raise NotImplementedError
+
+    @plugs.tag
+    def remove(self):
+        raise NotImplementedError
+
+    @plugs.tag
+    def move(self):
+        # song_id = param.get("song_id")
+        # step = param.get("step")
+        # MPDClient.playlistmove(name, from, to)
+        # MPDClient.swapid(song1, song2)
+        raise NotImplementedError
+
+    def test_mutex(self, delay):
+        self.mpd_mutex.acquire()
+        time.sleep(delay)
+        self.mpd_mutex.release()
+
+    @plugs.tag
+    def playsingle(self):
+        raise NotImplementedError
+
+    @plugs.tag
+    def resume(self):
+        songpos = self.current_folder_status["CURRENTSONGPOS"]
+        elapsed = self.current_folder_status["ELAPSED"]
+        self.mpd_retry_with_mutex(self.mpd_client.seek, songpos, elapsed)
+        self.mpd_retry_with_mutex(self.mpd_client.play)
+
+    @plugs.tag
+    def playlistaddplay(self, folder):
+        # add to playlist (and play)
+        # this command clears the playlist, loads a new playlist and plays it. It also handles the resume play feature.
+        logger.info(f"playing folder: {folder}")
+        self.mpd_retry_with_mutex(self.mpd_client.clear)
+
+        if folder is not None:
+            # TODO: why dealing with playlists? at least partially redundant with folder.config,
+            # so why not combine if needed alternative solution, just add folders recursively to quene
+            self.mpd_retry_with_mutex(self.mpd_client.add, folder)
+
+            self.music_player_status['player_status']['last_played_folder'] = folder
+
+            self.current_folder_status = self.music_player_status['audio_folder_status'].get(folder)
+            if self.current_folder_status is None:
+                self.current_folder_status = self.music_player_status['audio_folder_status'][folder] = {}
+
+            self.mpd_retry_with_mutex(self.mpd_client.play)
+
+        return self.mpd_status
+
+    @plugs.tag
+    def playerstatus(self):
+        return self.mpd_status
+
+    @plugs.tag
+    def playlistinfo(self):
+        playlistinfo = (self.mpd_retry_with_mutex(self.mpd_client.playlistinfo))
+        return playlistinfo
+
+    # Attention: MPD.listal will consume a lot of memory with large libs.. should be refactored at some point
+    @plugs.tag
+    def list_all_dirs(self):
+        result = self.mpd_retry_with_mutex(self.mpd_client.listall)
+        # list = [entry for entry in list if 'directory' in entry]
+        return result
```

Comment:
I tried using this in my webapp with this payload

```
{ package: "player", plugin: "ctrl", name: "ctrl", method: "list_all_dirs", kwargs: {} }
```

It works properly, the package and plugin gets called perfectly and the debugger returns the following

```
-   58:server.py       - jb.rpc.server        - MainThread      - DEBUG    - Request: {'package': 'player', 'plugin': 'ctrl', 'name': 'ctrl', 'method': 'list_all_dirs', 'kwargs': {}}
-  610:plugs.py        - jb.plugin.call       - MainThread      - DEBUG    - Calling: player.ctrl.list_all_dirs(args=(), kwargs={})
-  111:server.py       - jb.rpc.server        - MainThread      - DEBUG    - Sending response: {'result': None}
```

You can already see, the return result is `{'result': None}` , but I am actually expecting the list of actual directories. When inspecting `result` using the debugger, it does show the resultset of `listall` but it does not get passed on.

Reply:
Never mind, after reading your code more carefully, I saw the notion of `id` to be a required field. I remember you told us about it. It now works